### PR TITLE
Add support for visionOS

### PR DESCRIPTION
**Pull request by marcomasser**
This PR adds support for visionOS when compiling using Xcode 15 and Swift 5.9 but keeps compatibility with earlier versions of Xcode and Swift.

FYI: I did not update the Xcode project file because Swift Package Manager does not support generating that anymore because opening Swift packages is done by Xcode directly now.

### CODE REVIEW

File: Ashton.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -514,15 +514,19 @@
 				HEADER_SEARCH_PATHS = "";
 				INFOPLIST_FILE = Ashton.xcodeproj/Ashton_Info.plist;
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
+				IPHONEOS_DEPLOYMENT_TARGET = 13.4;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				MACOSX_DEPLOYMENT_TARGET = 10.15;
```
Comment (marcomasser) [inline Ashton.xcodeproj/project.pbxproj:519]: These are the same versions as in the `Package.swift` file.

File: Ashton.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -514,15 +514,19 @@
...
 				MODULEMAP_FILE = "";
 				PRODUCT_BUNDLE_IDENTIFIER = com.ideasoncanvas.Ashton;
 				PRODUCT_NAME = "$(TARGET_NAME:c99extidentifier)";
 				PROVISIONING_PROFILE_SPECIFIER = "";
 				SKIP_INSTALL = YES;
+				SUPPORTED_PLATFORMS = "iphoneos iphonesimulator macosx xros xrsimulator";
+				SUPPORTS_MACCATALYST = NO;
```
Comment (marcomasser) [inline Ashton.xcodeproj/project.pbxproj:526]: Xcode really wants to add this when adding visionOS.

File: Package.swift [hunk 3]
```diff
@@ -1,9 +1,9 @@
-// swift-tools-version:5.1
+// swift-tools-version:5.9
 import PackageDescription
 
 let package = Package(
     name: "Ashton",
-    platforms: [.iOS("13.4"), .macOS(.v10_15)],
+    platforms: [.iOS("13.4"), .macOS(.v10_15), .visionOS(.v1)],
```
Comment (marcomasser) [inline Package.swift:6]: Trying to use `.visionOS(…)` when the `swift-tools-version` is lower than `5.9` doesn’t seem to work 😞

**Comment by marcomasser** [comment]
I’m not sure why but when using this branch as a dependency in another Swift package, Xcode shows an error. Will investigate further.

File: Package@swift-5.9.swift [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// swift-tools-version:5.9
+import PackageDescription
+
+let package = Package(
+    name: "Ashton",
+    platforms: [.iOS("13.4"), .macOS(.v10_15), .visionOS(.v1)],
```
Comment (marcomasser) [inline Package@swift-5.9.swift:6]: This is a new file used only by Swift 5.9. Lower versions will keep using `Package.swift` and won’t see the visionOS platform.

File: Sources/Ashton/AshtonHTMLWriter.swift [hunk 5]
```diff
@@ -7,7 +7,7 @@
 //
 
 import Foundation
-#if os(iOS)
+#if os(iOS) || (compiler(>=5.9) && os(visionOS))
```
Comment (marcomasser) [inline Sources/Ashton/AshtonHTMLWriter.swift:10]: Writing it with the `compiler(>=5.9)` part prevents warnings about visionOS being an unknown platform when compiling using earlier versions of Swift.

**Review by marcomasser — COMMENTED** [review]
With the latest changes the Package builds without warnings using Xcode 14.3.1 and Xcode 15.0 Beta 8. I tested building for macOS and iOS using both versions of Xcode and building for visionOS using Xcode 15.

**Comment by marcomasser** [comment]
FYI: I did run the unit tests on all supported platforms and they do actually succeed 🙂

File: README.md [hunk 6]
```diff
@@ -64,8 +64,8 @@ github "IdeasOnCanvas/Ashton"
 
 ### Integration with the Swift Package Manager
 
-The Swift Package Manager is a dependency manager integrated with the Swift build system. To learn how to use the Swift Package Manager for your project, please read the [official documentation](https://github.com/apple/swift-package-manager/blob/master/Documentation/Usage.md).  
-To add ZIP Foundation as a dependency, you have to add it to the `dependencies` of your `Package.swift` file and refer to that dependency in your `target`.
+The Swift Package Manager is a dependency manager integrated with the Swift build system. To learn how to use the Swift Package Manager for your project, please read the [official documentation](https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md).  
+To add Ashton as a dependency, you have to add it to the `dependencies` of your `Package.swift` file and refer to that dependency in your `target`.
```
Comment (schwmi) [inline README.md:68]: :) good catch

**Review by schwmi — APPROVED** [review]
Change looks good - maybe you can also update the badge on the README page to add visionOS as additional supported platform…

**Comment by marcomasser** [comment]
> […] maybe you can also update the badge on the README page to add visionOS as additional supported platform…

I already did 🙂

Bildschirmfoto 2023-09-01 um 09 32 42

**Pull request merged**